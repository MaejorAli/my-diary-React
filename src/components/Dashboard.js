import React from 'react';
import { Link } from 'react-router-dom';


const Dashboard = () => (
  <div className="dashboard">
    <header>
      <div id="menu">
        <Link to="/dashboard" className="menuitem">Home</Link>
        <Link to="/profile" className="menuitem">Profile</Link>
        <Link to="/entries" className="menuitem">Entries</Link>
        <Link to="/signout" className="menuitem">Sign Out</Link>
      </div>
    </header>
    <div className="center">
      <h1>Hello,</h1>
      <h2>Its me your dear Diary</h2>
      <p>Feel free to pen down your thoughts
        <Link to="/entries/create"> <button type="button">Get Started</button></Link>
      </p>
    </div>
  </div>
);

export default Dashboard;
