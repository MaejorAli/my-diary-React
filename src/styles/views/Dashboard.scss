.dashboard {
  font-family: 'Roboto', sans-serif;
  background: #f4f4f4;
  background-image: url("../assets/indexBackground.jpg");
  background-size: cover;
  height: 700px;
  margin: 0px;
  #menu{
    text-align: center;
      font-weight: bold;    
    }
    .menuitem{
    padding: 10px;
      cursor: pointer;
      text-decoration: none;
      color: white;
      font-weight: 300;
      
      
      
    }
    .menuitem:hover{
    background-color: transparent;
      color: grey;
      
    }
    header {
      
    padding: 30px;
    text-align: center;
    background-color: #9099A2;
    }
    
    h2 {
    text-align: center;
    font-family: 'Roboto', sans-serif;
    font-weight: 500;
    }
    P {
    text-align: center;
    font-family: 'Roboto', sans-serif;
    }
    button{
    text-align: center;
    padding: 8px 10px;
    font-size: 16px;
    text-align: center;
    cursor: pointer;
    outline: none;
    color: #fff;
    background-color: #9099A2;
    border: none;
    border-radius: 15px;
    }
    .center {
    height: 200px;
    width: 400px;
    background: transparent;
    color: white;
    position: fixed;
    top: 50%;
    left: 50%;
    margin-top: 50px;
    margin-left: -200px;
    }
    
}

