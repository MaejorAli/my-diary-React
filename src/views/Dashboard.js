import React from 'react';
import Dashboard from '../components/Dashboard';


export default (props) => (
  <div>
    <Dashboard {...props} />
  </div>
);
