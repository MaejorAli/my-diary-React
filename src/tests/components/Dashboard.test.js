import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Dashboard from '../../components/Dashboard';

let wrapped;

beforeEach(() => {
  wrapped = shallow(<Dashboard />);
});


describe('Dashboard Component', () => {
  describe('render features', () => {
    test('view should render as expected', () => {
      const tree = toJson(wrapped);
      expect(tree).toMatchSnapshot();
    });
  });
});
