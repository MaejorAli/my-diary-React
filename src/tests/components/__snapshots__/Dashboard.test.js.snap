// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard Component render features view should render as expected 1`] = `
<div
  className="dashboard"
>
  <header>
    <div
      id="menu"
    >
      <Link
        className="menuitem"
        replace={false}
        to="/dashboard"
      >
        Home
      </Link>
      <Link
        className="menuitem"
        replace={false}
        to="/profile"
      >
        Profile
      </Link>
      <Link
        className="menuitem"
        replace={false}
        to="/entries"
      >
        Entries
      </Link>
      <Link
        className="menuitem"
        replace={false}
        to="/signout"
      >
        Sign Out
      </Link>
    </div>
  </header>
  <div
    className="center"
  >
    <h1>
      Hello,
    </h1>
    <h2>
      Its me your dear Diary
    </h2>
    <p>
      Feel free to pen down your thoughts
      <Link
        replace={false}
        to="/entries/create"
      >
         
        <button
          type="button"
        >
          Get Started
        </button>
      </Link>
    </p>
  </div>
</div>
`;
